onboard/src/pages/subsections/backgroundInformation/Personalnformation.jsx

```diff
@@ -1,63 +1,362 @@
+  const [phoneNum, setPhoneNum] = useState("");
+  const [secondaryPhoneNum, setSecondaryPhoneNum] = useState("");
+  const [ssn, setSSN] = useState("");
+  const [dateOfBirth, setDateOfBirth] = useState(new Date());
+  const [currentAddress, setCurrentAddress] = useState("");
+  const [permanentAddress, setPermanentAddress] = useState("");
+
+  //const [dateOfBirthHelper, setDateOfBirthHelper] = useState("");
+  const [firstNameHelper, setFirstNameHelper] = useState("");
+  const [lastNameHelper, setLastNameHelper] = useState("");
+  const [phoneNumHelper, setPhoneNumHelper] = useState("");
+  const [middleInitalHelper, setMiddleInitalHelper] = useState("");
+  const [secondaryPhoneNumHelper, setSecondaryPhoneNumHelper] = useState("");
+  const [ssnHelper, setSSNHelper] = useState("");
+
+  const [currentAddressHelper, setCurrentAddressHelper] = useState("");
+  const [permanentAddressHelper, setPermanentAddressHelper] = useState("");
+
+  //const [dateOfBirthError, setDateOfBirthError] = useState(false);
+  const [firstNameError, setFirstNameError] = useState(false);
+  const [lastNameError, setLastNameError] = useState(false);
+  const [middleInitalError, setMiddleInitalError] = useState(false);
+  const [phoneNumError, setPhoneNumError] = useState(false);
+  const [secondaryPhoneNumError, setSecondaryPhoneNumError] = useState(false);
+  const [ssnError, setSSNError] = useState(false);
+
+  const [currentAddressError, setCurrentAddressError] = useState(false);
+  const [permanentAddressError, setPermanentAddressError] = useState(false);
+
+
+  function handleChange(event) {
+    const e = event;
+    const val = e.target.value;
+    const name = e.target.name;
+
+    console.log(val);
+    console.log(name);
+
+    // Event handler for the fields
+
+    if (name === "firstName") {
+      setFirstName(val);
+      if (val.length >= 2) {
+        setFirstNameError(false);
+        setFirstNameHelper("");
+      } else {
+        setFirstNameError(true);
+        setFirstNameHelper("Please enter a full first name.");
+      }
+    } else if (name === "lastName") {
+      setLastName(val);
+      if (val.length >= 2) {
+        setLastNameError(false);
+        setLastNameHelper("");
+      } else {
+        setLastNameError(true);
+        setLastNameHelper("Please enter a full last name.");
+      }
+    } else if (name === "middleInitial") {
+      setMiddleInital(val);
+      if (val.length === 1) {
+        setMiddleInitalError(false);
+        setMiddleInitalHelper("");
+      } else {
+        setMiddleInitalError(true);
+        setMiddleInitalHelper("Please enter a one letter initial.");
+      }
+    } else if (name === "ssn") {
+      setSSN(val);
+      if (validSSN.test(val)) {
+        setSSNError(false);
+        setSSNHelper("");
+      } else {
+        setSSNError(true);
+        setSSNHelper("Please enter a valid SSN.");
+      }
+    } else if (name === "dob") {
+      setDateOfBirth(val);
```

Comment:
Add a regex validation for all date fields to check whether the value is the correct type of input & also to see if the date entered is within a reasonable timeframe for human life.